vcpkg_extract_source_archive fails on macOS when build tree is open in the finder

**Describe the bug**
When vcpkg is extracting the source on macOS the temporary extraction directory is checked to ensure only 1 file exists, and that it is a directory. If the build tree is opened in the finder the finder often creates a .DS_STORE file in the temp directory. This causes the build to fail, as a `.DS_STORE` file exists alongside the extracted source directory.

The build fails with:
```
CMake Error at scripts/cmake/vcpkg_extract_source_archive.cmake:225 (message):
  Could not unwrap top level directory from archive.  Pass
  NO_REMOVE_ONE_LEVEL to disable this.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_extract_source_archive_ex.cmake:32 (vcpkg_extract_source_archive)
  ports/qt5-base/cmake/qt_download_submodule.cmake:32 (vcpkg_extract_source_archive_ex)
  ports/qt5-base/portfile.cmake:72 (qt_download_submodule)
  scripts/ports.cmake:146 (include)
```

**Environment**
- macOS

**To Reproduce**
I noticed this while trying to install qt5-base, but its likely any port which uses the CMake function vcpkg_extract_source_archive would have the same issue. It does not always happen.

1. Open the vcpkg build tree directory in the finder (`<vcpkgRoot>/buildtrees`)
2. Run `vcpkg install qt5-base`

If you add `message("Temp directory: '${temp_dir}' Archive Dir: ${archive_directory}")` to line 221 of `scripts/cmake/vcpkg_extract_source_archive.cmake` and rerun the above steps, its evident that a `.DS_STORE` file is shown in the resulting glob.

--- a/scripts/cmake/vcpkg_extract_source_archive.cmake
+++ b/scripts/cmake/vcpkg_extract_source_archive.cmake
@@ -218,6 +218,8 @@ function(vcpkg_extract_source_archive)
         cmake_path(SET temp_source_path "${temp_dir}")
     else()
         file(GLOB archive_directory "${temp_dir}/*")
+        # Exclude .DS_Store entries created by the finder on macOS
+        list(FILTER archive_directory EXCLUDE REGEX ".*/.DS_Store$")
         # make sure `archive_directory` is only a single file
         if(NOT archive_directory MATCHES ";" AND IS_DIRECTORY "${archive_directory}")
             cmake_path(SET temp_source_path "${archive_directory}")

